drivers/soundwire/intel.c

```diff
@@ -401,6 +401,45 @@ static void intel_shim_wake(struct sdw_intel *sdw, bool wake_enable)
 	}
 }
 
+void sdw_intel_process_wakeen_event(struct sdw_intel_ctx *ctx)
+{
+	struct sdw_intel_link_res *link;
+	struct sdw_master_device *md;
+	struct sdw_slave *slave;
+	struct sdw_bus *bus;
+
+	list_for_each_entry(link, &ctx->link_list, list) {
+		struct sdw_intel *sdw;
+		u16 wake_sts;
+
+		md = link->md;
+		bus = &link->cdns->bus;
+		sdw = cdns_to_intel(link->cdns);
+		wake_sts = intel_readw(sdw->link_res->shim, SDW_SHIM_WAKESTS);
+
+		/* disable WAKEEN interrupt ASAP to prevent interrupt flood */
+		intel_shim_wake(sdw, false);
```

Comment:
@RanderWang just wondering: do the above 2  statements to read the wake_sts and disable wakeen need to be inside the list iterator or can they be outside?

Reply:
@ranj063  intel_shim_wake(sdw, false) is for each link,  so it can't be outside.
 Yes, It is better to move wake_sts outsize. I found it was easy to get shim with sdw->link_res->shim in inside.

 I will move it outside.